### PR TITLE
Template rendering with preview

### PR DESCRIPTION
**Pull request by ares**
based on https://github.com/theforeman/foreman_remote_execution/pull/6

### CODE REVIEW

File: app/controllers/job_templates_controller.rb [hunk 1]
```diff
@@ -7,7 +7,19 @@ def load_vars_from_template
   end
 
   def auto_complete_job_name
-    @job_names = resource_base.where(['job_name LIKE ?', "%#{params[:search]}%"]).pluck(:job_name)
+    @job_names = resource_base.where(['job_name LIKE ?', "%#{params[:search]}%"]).pluck(:job_name).uniq
```
Comment (ohadlevy) [inline app/controllers/job_templates_controller.rb]: afair quoting around params is not needed, as where will do it for you.

Comment (ares) [inline app/controllers/job_templates_controller.rb]: not sure I follow, I need to have % as part of that string that's injected

**Comment by ares** [comment]
For testing you need to apply https://github.com/theforeman/foreman/pull/2524 to your Foreman instance

**Comment by ares** [comment]
After core PR is merged I'll check compatibility with 1.9 and potentially add some version detection so this is active only in 1.10 if possible.

**Comment by ares** [comment]
It seems compatible with 1.9 (just the preview button is missing) so we can safely merge to master branch.

**Comment by ares** [comment]
Works for me with lastest 1.9-stable and is required for compatibility with 1.10 (otherwise job template form fails because of unknown preview route) so this should get in ASAP. @iNecas could you review?

**Comment by iNecas** [comment]
I've tested this with https://github.com/theforeman/foreman_remote_execution/pull/12 against 1.9, as well as foreman develop (with https://github.com/theforeman/foreman/pull/2591), and worked as expected, one thing that I've noticed is that we don't accept the `@host` variable, therefore we can't use the things like hostname in the template, but I will open separate PR to fix that. ACK

**Pull request merged**